addons/udes_stock/models/stock_move.py

```diff
@@ -146,8 +146,13 @@ def split_out_move_lines(self, move_lines):
     def _action_assign(self):
         res = super(StockMove, self)._action_assign()
         for picking_type in self.mapped('picking_type_id'):
-            self.filtered(lambda m: m.picking_type_id == picking_type). \
-                post_reservation_split()
+            moves = self.filtered(lambda m: m.picking_type_id == picking_type)
+            moves.post_reservation_split()
+            if picking_type.u_preprocess_drop_location:
+                moves.mapped('picking_id').apply_drop_location_policy()
```

Comment:
Question: as we are iterating the picking types and, based on that, filtering the move lines, we can be sure that the `apply_drop_location_policy` logic (such method iterates all the recordset instances) won't be applied to the same picking multiple times, right?

Reply:
moves.mapped('picking_id') will de-duplicate the pickings and a picking only has one picking_type so yeah it will only be applied once per pick.